File: app/src/main/java/adapter/ElencoAdapter.java
package adapter;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.squareup.picasso.MemoryPolicy;
import com.squareup.picasso.NetworkPolicy;
import com.squareup.picasso.Picasso;

import java.util.List;

import activity.PersonActivity;
import br.com.icaro.filme.R;
import info.movito.themoviedbapi.model.people.PersonCast;
import utils.Constantes;
import utils.UtilsApp;

/**
 * Created by icaro on 24/07/16.
 */
public class ElencoAdapter extends RecyclerView.Adapter<ElencoAdapter.ElencoViewHolder> {

    private final FirebaseAnalytics mFirebaseAnalytics;
    private Context context;
    private List<PersonCast> casts;


    public ElencoAdapter(Context elencoActivity, List<PersonCast> casts) {

        this.context = elencoActivity;
        this.casts = casts;
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(elencoActivity);
       // Log.d("ElencoAdapter", "Tamanho " + casts.size());
    }

    @Override
    public ElencoAdapter.ElencoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.elenco_list_adapter, parent, false);
        return new ElencoViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ElencoAdapter.ElencoViewHolder holder, int position) {
        final PersonCast personCast = casts.get(position);
        holder.elenco_character.setText(personCast.getCharacter());

        holder.elenco_nome.setText(personCast.getName());
        Picasso.with(context).load(UtilsApp
                .getBaseUrlImagem(UtilsApp.getTamanhoDaImagem(context, 2)) + personCast.getProfilePath())
                .placeholder(R.drawable.person)
                .memoryPolicy(MemoryPolicy.NO_STORE, MemoryPolicy.NO_CACHE)
                .networkPolicy(NetworkPolicy.NO_CACHE, NetworkPolicy.NO_STORE)
                .into(holder.img_elenco);


        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Bundle bundle = new Bundle();
                bundle.putString(FirebaseAnalytics.Param.ITEM_ID, String.valueOf(personCast.getId()));
                bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, personCast.getName());
                mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SELECT_CONTENT, bundle);

                Intent intent = new Intent(context, PersonActivity.class);
                intent.putExtra(Constantes.INSTANCE.getPERSON_ID(), personCast.getId());
                intent.putExtra(Constantes.INSTANCE.getNOME_PERSON(), personCast.getName());
                context.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
      //  Log.d("ElencoAdapter", "Tamanho " + casts.size());
        if (casts != null){
            return  casts.size();
        } else {
            return 0;
        }
    }

    class ElencoViewHolder extends RecyclerView.ViewHolder {

        private TextView elenco_nome, elenco_character;
        private ImageView img_elenco;

        ElencoViewHolder(View itemView) {
            super(itemView);
            elenco_nome = (TextView) itemView.findViewById(R.id.elenco_nome);
            elenco_character = (TextView) itemView.findViewById(R.id.elenco_character);
            img_elenco = (ImageView) itemView.findViewById(R.id.img_elenco);
        }
    }
}

File: app/src/main/java/elenco/ElencoAdapter.kt
package elenco

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.com.icaro.filme.R
import domain.CastItem
import pessoa.activity.PersonActivity
import utils.Constant
import utils.setPicassoWithCache

/**
 * Created by icaro on 24/07/16.
 */
class ElencoAdapter(private val context: Context, private val casts: List<CastItem?>?) : RecyclerView.Adapter<ElencoAdapter.ElencoViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ElencoViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.elenco_list_adapter, parent, false)
        return ElencoViewHolder(view)
    }

    override fun onBindViewHolder(holder: ElencoViewHolder, position: Int) {
        val (_, character, _, _, name, profilePath, id) = casts?.get(position)!!
        holder.elencoCharacter.text = character

        holder.elencoNome.text = name

        holder.imgElenco.setPicassoWithCache(profilePath, 2, img_erro = R.drawable.person)

        holder.itemView.setOnClickListener {
            context.startActivity(Intent(context, PersonActivity::class.java).apply {
                putExtra(Constant.PERSON_ID, id)
                putExtra(Constant.NOME_PERSON, name)
            })
        }
    }

    override fun getItemCount(): Int {
        return casts?.size ?: 0
    }

    inner class ElencoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val elencoNome: TextView = itemView.findViewById(R.id.elenco_nome)
        val elencoCharacter: TextView = itemView.findViewById(R.id.elenco_character)
        val imgElenco: ImageView = itemView.findViewById(R.id.img_elenco)
    }
}
